src/main/scala/io/github/ilyazinovich/dmmf/PriceOrder.scala

```diff
@@ -22,20 +25,14 @@ object PriceOrder {
     val pricedLines: List[Either[Error, PricedOrderLine]] = orderLines.map {
       case OrderLine(orderLineId, productCode, quantity) =>
         val orderLinePrice = Price.multiply(getProductPrice(productCode), quantity)
-        orderLinePrice.right.map(p => PricedOrderLine(orderLineId, productCode, quantity, p))
+        orderLinePrice.map(p => PricedOrderLine(orderLineId, productCode, quantity, p))
     }
     val accumulator: Either[NonEmptyList[Error], List[PricedOrderLine]] = Right(List.empty)
-    pricedLines.foldRight(accumulator)(reduce)
+    pricedLines.foldLeft(accumulator)(reduce)
   }
 
-  private def reduce(orderLinePriceCalculationResult: Either[Error, PricedOrderLine],
-                     accumulator: Either[NonEmptyList[Error], List[PricedOrderLine]]) = {
-    (orderLinePriceCalculationResult, accumulator) match {
-      case (Left(error), Left(errorList)) => Left(error :: errorList)
-      case (Left(error), Right(_)) => Left(NonEmptyList.of(error))
-      case (Right(_), Left(errorList)) => Left(errorList)
-      case (Right(pricedOrderLine), Right(pricedOrderLines)) =>
-        Right(pricedOrderLine :: pricedOrderLines)
-    }
+  private def reduce(accumulator: Either[NonEmptyList[Error], List[PricedOrderLine]],
+                     orderLinePriceCalculationResult: Either[Error, PricedOrderLine]) = {
+    orderLinePriceCalculationResult.bimap(NonEmptyList.one, List(_)) |+| accumulator
```

Comment:
Let's better use less scary `combine` function instead of `|+|` 🤓

Reply:
Should I have an abstract factory for that?

Reply:
no, `combine` is just fine